pkg/server/auth/server.go

```diff
@@ -286,33 +293,50 @@ func (s *AuthServer) SignIn() http.HandlerFunc {
 			return
 		}
 
-		var hashedSecret corev1.Secret
+		if loginRequest.Username == "" || loginRequest.Password == "" {
+			http.Error(rw, "invalid credentials", http.StatusBadRequest)
+			return
+		}
 
-		if err := s.kubernetesClient.Get(r.Context(), ctrlclient.ObjectKey{
-			Namespace: v1alpha1.DefaultNamespace,
-			Name:      ClusterUserAuthSecretName,
-		}, &hashedSecret); err != nil {
-			s.Log.Error(err, "Failed to query for the secret")
-			http.Error(rw, "Please ensure that a password has been set.", http.StatusBadRequest)
+		users := &corev1.SecretList{}
 
-			return
+		opts := []client.ListOption{
+			GitopsUserMatchingLabel,
+			client.InNamespace(wego.DefaultNamespace),
 		}
 
-		if loginRequest.Username != string(hashedSecret.Data["username"]) {
-			s.Log.Info("Wrong username")
-			rw.WriteHeader(http.StatusUnauthorized)
+		if err := s.kubernetesClient.List(r.Context(), users, opts...); err != nil {
```

Comment:
we can't query based on any data value. So to accomplish that we would need to force users to use some kind of convention for names, but that would still be tricky because they can choose to use an email address as the username and that's not a valid name for a k8s resource.

Listing isn't ideal but at least we only list the secrets with that particular Label. And in practice, there should be a small number of those.

Reply:
aah i see, makes sense